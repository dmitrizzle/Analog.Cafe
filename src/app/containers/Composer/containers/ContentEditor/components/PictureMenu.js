// tools
import React from "react"
import styled from "styled-components"

// components
import {
  ButtonStrip,
  Item
} from "../../../../../components/_controls/ButtonStrip"

const PictureMenu = styled.div`
  width: 100%;
  position: absolute;
  left: 0;
  margin-top: ${props => props.theme.size.block.spacing}em;
  z-index: ${props => props.theme.layer.up + 2};

  & a:first-child {
    border-right: 1px solid
      ${props => props.theme.color.alpha.background(props.theme.opacity.least)};
  }
`

// return
export default props => {
  return (
    <PictureMenu style={{ display: props.focus ? "block" : "none" }}>
      <ButtonStrip style={{ margin: "0 auto" }}>
        <div>
          <Item
            onClick={event => {
              event.preventDefault()
              event.stopPropagation()
              props.focus && props.removePicture()
            }}
            left
            black
          >
            Delete
          </Item>
          <Item
            onClick={event => {
              event.preventDefault()
              event.stopPropagation()
              props.focus && props.featurePicture()
            }}
            title="âŒ˜ + F"
            right
            black
          >
            Resize
          </Item>
        </div>
      </ButtonStrip>
    </PictureMenu>
  )
}
