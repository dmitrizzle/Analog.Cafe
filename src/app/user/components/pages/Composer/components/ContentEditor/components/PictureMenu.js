import { ButtonStrip } from "@roast-cms/react-button-beans"
import React from "react"
import styled from "styled-components"

import ButtonStripItem from "../../../../../../../core/components/controls/Button/components/ButtonStripItem"

// components

const PictureMenu = styled.div`
  width: 100%;
  position: absolute;
  left: 0;
  margin-top: ${props => props.theme.size.block.spacing}em;
  z-index: ${props => props.theme.layer.up + 2};

  & a:first-child {
    border-right: 1px solid
      ${props => props.theme.color.background(props.theme.opacity.least)};
  }
`

// return
export default props => {
  return (
    <PictureMenu style={{ pointerEvents: "none" }}>
      <ButtonStrip style={{ margin: "0 auto", pointerEvents: "auto" }}>
        <div>
          <ButtonStripItem
            onMouseDown={event => {
              event.preventDefault()
              props.removePicture()
            }}
            left
            inverse
          >
            Delete
          </ButtonStripItem>
          <ButtonStripItem
            onMouseDown={event => {
              event.preventDefault()
              props.featurePicture()
            }}
            title="âŒ˜ + F"
            right
            inverse
          >
            Resize
          </ButtonStripItem>
        </div>
      </ButtonStrip>
    </PictureMenu>
  )
}
