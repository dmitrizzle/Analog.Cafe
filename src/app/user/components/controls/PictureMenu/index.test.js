import { shallow } from "enzyme"
import React from "react"

import PictureMenu from "./"

it("Render PictureMenu without crashing", () => {
  shallow(<PictureMenu />)
})
