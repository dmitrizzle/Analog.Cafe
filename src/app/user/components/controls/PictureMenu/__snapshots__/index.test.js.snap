// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PictureMenu, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        style={
                Object {
                        "margin": "0 auto",
                        "pointerEvents": "auto",
                      }
        }
>
        <div>
                <Unknown
                        inverse={true}
                        left={true}
                        onMouseDown={[Function]}
                >
                        Delete
                </Unknown>
                <Unknown
                        inverse={true}
                        onMouseDown={[Function]}
                        right={true}
                        title="⌘ + F"
                >
                        Resize
                </Unknown>
        </div>
</styled.div>,
      "style": Object {
        "pointerEvents": "none",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Unknown
                    inverse={true}
                    left={true}
                    onMouseDown={[Function]}
          >
                    Delete
          </Unknown>
          <Unknown
                    inverse={true}
                    onMouseDown={[Function]}
                    right={true}
                    title="⌘ + F"
          >
                    Resize
          </Unknown>
</div>,
        "style": Object {
          "margin": "0 auto",
          "pointerEvents": "auto",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              inverse={true}
              left={true}
              onMouseDown={[Function]}
>
              Delete
</Unknown>,
            <Unknown
              inverse={true}
              onMouseDown={[Function]}
              right={true}
              title="⌘ + F"
>
              Resize
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Delete",
              "inverse": true,
              "left": true,
              "onMouseDown": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Resize",
              "inverse": true,
              "onMouseDown": [Function],
              "right": true,
              "title": "⌘ + F",
            },
            "ref": null,
            "rendered": "Resize",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          style={
                    Object {
                              "margin": "0 auto",
                              "pointerEvents": "auto",
                            }
          }
>
          <div>
                    <Unknown
                              inverse={true}
                              left={true}
                              onMouseDown={[Function]}
                    >
                              Delete
                    </Unknown>
                    <Unknown
                              inverse={true}
                              onMouseDown={[Function]}
                              right={true}
                              title="⌘ + F"
                    >
                              Resize
                    </Unknown>
          </div>
</styled.div>,
        "style": Object {
          "pointerEvents": "none",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <Unknown
                        inverse={true}
                        left={true}
                        onMouseDown={[Function]}
            >
                        Delete
            </Unknown>
            <Unknown
                        inverse={true}
                        onMouseDown={[Function]}
                        right={true}
                        title="⌘ + F"
            >
                        Resize
            </Unknown>
</div>,
          "style": Object {
            "margin": "0 auto",
            "pointerEvents": "auto",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                inverse={true}
                left={true}
                onMouseDown={[Function]}
>
                Delete
</Unknown>,
              <Unknown
                inverse={true}
                onMouseDown={[Function]}
                right={true}
                title="⌘ + F"
>
                Resize
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Delete",
                "inverse": true,
                "left": true,
                "onMouseDown": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Resize",
                "inverse": true,
                "onMouseDown": [Function],
                "right": true,
                "title": "⌘ + F",
              },
              "ref": null,
              "rendered": "Resize",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
